api/api.go

```diff
@@ -359,15 +360,33 @@ func (api *API) delSilence(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *API) listSilences(w http.ResponseWriter, r *http.Request) {
-	sils, err := api.silences.All()
+	var (
+		num    = r.FormValue("n")
+		offset = r.FormValue("offset")
```

Comment:
That was my original approach with `lastID`, but then I wasn't sure how to handle users skipping ahead several pages in the pagination. I originally though it unlikely, but then moved towards calculating the next to be shown using `num * offset`. It's easy to re-introduce, but was unsure about supporting jumping forward `1+n` pages, and opted for that in this case. Happy to change it back if that's the vote.

As you pointed out, since it's not a relational db the offset is quick because it jumps to a position in a slice.